crates/sui-display/src/v2/value.rs

```diff
@@ -187,10 +221,47 @@ impl Value<'_> {
             Value::String(s) => write!(w, "{s}")?,
 
             Value::Slice(s) => Formatter::deserialize_slice(*s, w)?,
+
+            Value::Bytes(_) => {
+                return Err(FormatError::TransformInvalid(
+                    "bytes cannot be formatted as a string",
+                ))
+            }
+
+            Value::Enum(_) => {
+                return Err(FormatError::TransformInvalid(
+                    "enums cannot be formatted as a string",
+                ))
+            }
+
+            Value::Struct(_) => {
+                return Err(FormatError::TransformInvalid(
+                    "struct literals cannot be formatted as a string",
+                ))
+            }
+
+            Value::Vector(_) => {
+                return Err(FormatError::TransformInvalid(
+                    "vector literals cannot be formatted as a string",
+                ))
+            }
         }
 
         Ok(())
     }
+
+    /// Coerce the value into a number, interpreted as an offset in milliseconds since the Unix
+    /// epoch, and format it as an ISO8601 timestamp.
+    fn format_as_timestamp(&self, w: &mut BoundedWriter<'_>) -> Result<(), FormatError> {
+        let ts = self
+            .as_u64()
+            .and_then(|ts| ts.try_into().ok())
+            .and_then(DateTime::from_timestamp_millis)
+            .ok_or_else(|| FormatError::TransformInvalid("not a timestamp"))?;
+
+        write!(w, "{ts:?}")?;
```

Comment:
Should probably not use debug format here, and either display format or go to the specific RFC format display you want here.

Reply:
This is a `chrono::DateTime` thing -- its Debug format is ISO8601.